### PR TITLE
[SPARK-3850] Turn style checker on for trailing whitespaces.

### PR DESCRIPTION
**Pull request by rxin**

### CODE REVIEW

File: scalastyle-config.xml [hunk 1]
```diff
@@ -50,6 +50,9 @@
  */]]></parameter>
   </parameters>
  </check>
+
+  <check level="error" class="org.scalastyle.file.WhitespaceEndOfLineChecker" enabled="true"></check>
```
Comment (andrewor14) [inline scalastyle-config.xml]: indent is weird

Comment (rxin) [inline scalastyle-config.xml]: I have a separate thing to rewrite this whole file later.

**Comment by rxin** [comment]
Jenkins, retest this please.

**Comment by rxin** [comment]
Actually I'm going to merge this since it's already passed the style checker stage.

File: scalastyle-config.xml (hunk 1 above)
Comment (rxin) [inline scalastyle-config.xml]: Here you go

https://github.com/apache/spark/pull/6543

**Pull request closed without merging**